core/src/main/java/org/veriblock/sdk/blockchain/store/VeriBlockStore.java

```diff
@@ -79,6 +79,10 @@ public StoredVeriBlockBlock setChainHead(StoredVeriBlockBlock chainHead) throws
     }
 
     public void put(StoredVeriBlockBlock storedBlock) throws BlockStoreException, SQLException {
+        if (veriBlockRepository.get(storedBlock.getHash()) != null) {
+            throw new BlockStoreException("A block with the same hash is already in the store");
```

Comment:
this should not be an error. 
```suggestion
            return;
```

Reply:
Return would hide the fact that the block didn't end up in the store. Thus, we would have data loss in the API except that instead of losing the block in the store, we would be losing the block we passed to put().

Reply:
we will not have "data loss", because this block already exists in chain.
imagine a situation when we have block 5 in store (as last block), and we put blocks from list [5,6,7,8];
when you silently return, blocks 6,7,8 end up in a store, but when you throw, blocks 6,7,8 will not

Reply:
The block may have and in our case does have additional metadata in the store, such as the block of proof. That's why replace() exists and is not a no-op.

The situation you describe is poorly-written code. Throwing an exception makes the API user aware of this, and forces them to either handle the exception or use replace() and hopefully check the return value.